lib/src/model/container.dart

```diff
@@ -268,6 +268,15 @@ abstract class Container extends ModelElement
   @override
   String get filePath => '${library.dirName}/$fileName';
 
+  /// The full path of this element's sidebar file.
+  String get sidebarPath;
```

Comment:
Containers have a notion of their own sidebar path (their "below" sidebar path), and then their child elements can call `enclosingElement.sidebarPath` for their aboveSidebarPaths.